project_stock/models/project_task.py

```diff
@@ -118,10 +118,7 @@ def _prepare_procurement_group_vals(self, values):
         return {"name": values["name"]}
 
     def action_confirm(self):
-        self.mapped("move_ids").write({"state": "confirmed"})
-        self.action_assign()
-
-    def action_assign(self):
+        self.mapped("move_ids")._action_confirm()
         self.mapped("move_ids")._action_assign()
```

Comment:
It seems I'm not able to transfer you what I mean. Let me put the code:

```suggestion
    def action_confirm(self):
        self.action_assign()

    def action_assign(self):
        self.action_confirm()
        self.mapped("move_ids")._action_assign()
```